assets/js/group-details.js

```diff
@@ -0,0 +1,467 @@
+/* global jQuery:false, wpApiSettings:false */
+
+jQuery.ajaxSetup({
+  beforeSend: function(xhr) {
+    xhr.setRequestHeader('X-WP-Nonce', wpApiSettings.nonce);
+  },
+})
+
+jQuery(document).ready(function($) {
+function save_field_api(groupId, post_data){
+  return jQuery.ajax({
+    type:"POST",
+    data:JSON.stringify(post_data),
+    contentType: "application/json; charset=utf-8",
+    dataType: "json",
+    url: wpApiSettings.root + 'dt-hooks/v1/group/'+ groupId,
+  })
+}
+
+function get_group(groupId){
+  return jQuery.ajax({
+    type:"GET",
+    contentType: "application/json; charset=utf-8",
+    dataType: "json",
+    url: wpApiSettings.root + 'dt-hooks/v1/group/'+ groupId
+  })
+}
+
+function add_item_to_field(groupId, post_data) {
+  return jQuery.ajax({
+    type: "POST",
+    data: JSON.stringify(post_data),
+    contentType: "application/json; charset=utf-8",
+    dataType: "json",
+    url: wpApiSettings.root + 'dt-hooks/v1/group/' + groupId + '/details',
+  })
+}
+
+function remove_item_from_field(groupId, fieldKey, valueId) {
+  let data = {key: fieldKey, value: valueId}
+  return jQuery.ajax({
+    type: "DELETE",
+    data: JSON.stringify(data),
+    contentType: "application/json; charset=utf-8",
+    dataType: "json",
+    url: wpApiSettings.root + 'dt-hooks/v1/group/' + groupId + '/details',
+  })
+}
+
+function post_comment(groupId, comment) {
+  return jQuery.ajax({
+    type: "POST",
+    data: JSON.stringify({comment}),
+    contentType: "application/json; charset=utf-8",
+    dataType: "json",
+    url: wpApiSettings.root + 'dt-hooks/v1/group/' + groupId + '/comment',
+  })
+}
+
+function get_comment(groupId) {
+  return jQuery.ajax({
+    type: "GET",
+    contentType: "application/json; charset=utf-8",
+    dataType: "json",
+    url: wpApiSettings.root + 'dt-hooks/v1/group/' + groupId + '/comments',
+  })
+}
+
+function get_activity() {
+  return jQuery.ajax({
+    type: "GET",
+    contentType: "application/json; charset=utf-8",
+    dataType: "json",
+    url: wpApiSettings.root + 'dt-hooks/v1/group/' + groupId + '/activity',
+  })
+}
+/**
+ * Typeahead functions
+ */
+
+function add_typeahead_item(groupId, fieldId, val, name) {
+  add_item_to_field(groupId, { [fieldId]: val }).done(function (addedItem){
+    jQuery(`.${fieldId}-list`).append(`<li class="${addedItem.ID}">
+    <a href="${addedItem.permalink}">${_.escape(addedItem.post_title)}</a>
+    <button class="details-remove-button details-edit"
+            data-field="locations" data-id="${val}"
+            data-name="${name}"  
```

Comment:
This needs escaping, you don't want to fail if `name` is `"><script src="http://example.com/malicious.js"></script>`